service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/CloneControlledAzureDatabaseResourceFlightTest.java

```diff
@@ -0,0 +1,168 @@
+import bio.terra.stairway.FlightStatus;
+import bio.terra.workspace.common.BaseAzureConnectedTest;
+import bio.terra.workspace.common.StairwayTestUtils;
+import bio.terra.workspace.common.fixtures.ControlledAzureResourceFixtures;
+import bio.terra.workspace.common.utils.AzureTestUtils;
+import bio.terra.workspace.connected.UserAccessUtils;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.database.AzureDatabaseUtilsRunner;
+import bio.terra.workspace.service.resource.controlled.flight.clone.azure.common.ClonedAzureResource;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import java.time.Duration;
+import java.util.UUID;
+import org.junit.jupiter.api.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Tag("azureConnected")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CloneControlledAzureDatabaseResourceFlightTest extends BaseAzureConnectedTest {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(CloneControlledAzureDatabaseResourceFlightTest.class);
+
+  @Autowired private JobService jobService;
+  @Autowired private AzureTestUtils azureTestUtils;
+  @Autowired private WorkspaceService workspaceService;
+  @Autowired private UserAccessUtils userAccessUtils;
+  @Autowired private AzureDatabaseUtilsRunner azureDatabaseUtilsRunner;
+
+  private Workspace sourceWorkspace;
+  private Workspace destinationWorkspace;
+
+  @BeforeAll
+  public void setup() throws InterruptedException {
+    sourceWorkspace =
+        createWorkspaceWithCloudContext(workspaceService, userAccessUtils.defaultUserAuthRequest());
+    destinationWorkspace =
+        createWorkspaceWithCloudContext(workspaceService, userAccessUtils.defaultUserAuthRequest());
+  }
+
+  @AfterAll
+  public void cleanup() {
+    // Deleting the workspace will also delete any resources contained in the workspace, including
+    // VMs and the resources created during setup.
+    workspaceService.deleteWorkspace(sourceWorkspace, userAccessUtils.defaultUserAuthRequest());
+    workspaceService.deleteWorkspace(
+        destinationWorkspace, userAccessUtils.defaultUserAuthRequest());
+  }
+
+  @Test
+  void cloneControlledAzureDatabaseFlightTest() throws InterruptedException {
+    FlightMap inputs = new FlightMap();
+
+    // To fill in this value:
+    // Go to your workspace dashboard -> cloud information -> copy the "Storage SAS URL"
+    inputs.put("BLOB_CONTAINER_URL_AUTHENTICATED", "https://...");
```

Comment:
You'll need to fill in this value to make this test work. There's no `WorkspaceFlightMapKeys` entry for this because we shouldn't be passing the SAS token around via the flight map. It should be generated on the fly.